### PR TITLE
refactor(log): use writeable options class to update config file

### PR DESCRIPTION
**Pull request by calahatras**
move option for spaces to view options in log page
use converter instead of separate flags for file view options in log page
GitDiffControl reacts to changes to ShowSpacesAsDots and rerenders file diff if selected diff is text changes

### CODE REVIEW

File: GitOut/Features/Git/Log/GitLogPage.xaml [hunk 1]
```diff
@@ -425,9 +442,25 @@
                         Placement="Bottom"
                     >
                         <Grid>
-                            <Border Style="{StaticResource CardPanelStyle}" MinWidth="180" Margin="16 8" Padding="0">
+                            <Grid.Resources>
+                                <Style x:Key="PopoverCheckboxListItemStyle" TargetType="{x:Type CheckBox}" BasedOn="{StaticResource MaterialCheckBoxStyle}">
+                                    <Setter  Property="Margin" Value="0 8" />
```
Comment (mickedefault) [inline GitOut/Features/Git/Log/GitLogPage.xaml]: adjust the two spaces to one

Comment (calahatras) [reply GitOut/Features/Git/Log/GitLogPage.xaml]: Fixed the spaces and also updated the spacing for self-closing elements to be the same (without space)

File: GitOut/Features/Options/OptionsWriter.cs [hunk 2]
```diff
@@ -0,0 +1,31 @@
+?using System;
```
Comment (mickedefault) [inline GitOut/Features/Options/OptionsWriter.cs]: is.. is that a BOM?

Comment (calahatras) [reply GitOut/Features/Options/OptionsWriter.cs]: da freq?

Comment (calahatras) [reply GitOut/Features/Options/OptionsWriter.cs]: :fire: faahr

**Review by mickedefault — APPROVED** [review]
:star:

**Pull request merged**